### PR TITLE
Implementing Zombie Runner

### PR DESCRIPTION
**Pull request by JamesPuli**
Implementation of Wandering Husk entity.
Known issues:
- Entity sometimes flickers when turning around on client side
- Entity does not make sound when "lunging" on client side.

### CODE REVIEW

File: HKMP/Game/Client/Entity/WanderingHusk.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
+using System.Collections;
```
Comment (Extremelyd1) [inline HKMP/Game/Client/Entity/WanderingHusk.cs:1]: I believe both this and the using directive below it are not used.

File: HKMP/Game/Client/Entity/WanderingHusk.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+        public override void UpdateAnimation(byte animationIndex, byte[] animationInfo) {
+            base.UpdateAnimation(animationIndex, animationInfo);
+
+            var animation = (Animation)animationIndex;
+
+            _lastAnimation = animation;
+
+            if (animation == Animation.Anticipate) {
+                _fsm.ExecuteActions("Anticipate", 2, 3, 5);
```
Comment (Extremelyd1) [inline HKMP/Game/Client/Entity/WanderingHusk.cs:105]: A few of the actions that you try to execute in this call are not implement in `ActionExtensions` in the `HKMP.Fsm` namespace. Therefore, they will not be executed and give a warning in the ModLog. I have implemented these actions in the `entity-system` branch, so after merging it they should work. This should also resolve the lunging sound not working client side.

Comment (Extremelyd1) [reply HKMP/Game/Client/Entity/WanderingHusk.cs:105]: As a heads up for next time: every time you see such an error in the ModLog, feel free to implement the action in `ActionExtensions` similarly to the others that are already present.

File: HKMP/Game/Client/Entity/WanderingHusk.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+using Hkmp.Util;
+using UnityEngine;
+
+namespace Hkmp.Game.Client.Entity {
+    public class WanderingHusk : HealthManagedEntity {
+        private readonly PlayMakerFSM _fsm;
+
+        private Animation _lastAnimation;
+
+        private Walker _walker;
```
Comment (Extremelyd1) [inline HKMP/Game/Client/Entity/WanderingHusk.cs:14]: This field and the `_audioSource` field can be made `readonly`.

**Review by Extremelyd1 — CHANGES_REQUESTED** [review]
Well done on the implementation! I have left a few minor remarks that should be easily solvable. 
Other than that, the `WanderingHusk.cs` file is not added to the `HKMP.csproj` and therefore after pulling your changes, the file wasn't noticed by my IDE for compilation. I'm fairly sure any IDE will modify the project file whenever a new file is added, so make sure you also push it with the changes!

File: HKMP/Game/Client/Entity/WanderingHusk.cs (hunk 1 above)
Comment (JamesPuli) [reply HKMP/Game/Client/Entity/WanderingHusk.cs]: My mistake, they were both unused. This should now be resolved.

File: HKMP/Game/Client/Entity/WanderingHusk.cs (hunk 2 above)
Comment (JamesPuli) [reply HKMP/Game/Client/Entity/WanderingHusk.cs:103]: Thanks for the heads up! I'll be sure to pay more attention for my next entity

File: HKMP/Game/Client/Entity/WanderingHusk.cs (hunk 3 above)
Comment (JamesPuli) [reply HKMP/Game/Client/Entity/WanderingHusk.cs]: I agree. This has been resolved.

**Review by Extremelyd1 — APPROVED** [review]

**Pull request merged**